app/models/user.rb

```diff
@@ -1,6 +1,9 @@
 class User < ApplicationRecord
+  has_many :authored_tests, class_name: 'Test', dependent: :nullify
+  has_many :user_tests,  dependent: :destroy
+  has_many :tests, through: :user_tests
+
   def get_tests_by_level(level)
-    Test.joins('JOIN user_tests ON tests.id = user_tests.test_id')
-        .where(level: level, user_tests: {user_id: id}) 
+    tests.select { |test| test.level == level } 
```

Comment:
```suggestion
    tests..where(level: level)
```
бд лучше справляется с селектами, к тому же не придется тянуть лишние записи из базы